File: Java/SpaceVIL/spacevil/src/main/java/com/spvessel/spacevil/Core/MouseArgs.java
package com.spvessel.spacevil.Core;

import java.util.LinkedList;
import java.util.List;

import com.spvessel.spacevil.Flags.InputState;
import com.spvessel.spacevil.Flags.KeyMods;
import com.spvessel.spacevil.Flags.MouseButton;

/**
 * A class that describes mouse inputs.
 */
public final class MouseArgs implements IInputEventArgs {
    /**
     * Mouse button as com.spvessel.spacevil.Flags.MouseButton.
     */
    public MouseButton button;

    /**
     * State of input as com.spvessel.spacevil.Flags.InputState. Values: RELEASE, PRESS, REPEAT.
     */
    public InputState state;

    /**
     * Used modifiers while mouse input.
     */
    public List<KeyMods> mods;

    /**
     * Mouse cursor position.
     */
    public Position position = new Position();

    public ScrollValue scrollValue = new ScrollValue();

    /**
     * Clearing MouseArgs
     */
    public void clear() {
        button = MouseButton.Unknown;
        state = InputState.Release;
        mods = new LinkedList<>();
        mods.add(KeyMods.No);
        position.setPosition(0, 0);
        scrollValue.setValues(0, 0);
    }

    @Override
    public String toString() {
        return
            button + " " +
            state + " " +
            mods + " " +
            scrollValue + " " +
            position.getX() + " " +
            position.getY() + " ";
    }
}

File: CSharp/SpaceVIL/SpaceVIL/Core/MouseArgs.cs
using System;

namespace SpaceVIL.Core
{
    /// <summary>
    /// A class that describes mouse inputs.
    /// </summary>
    public sealed class MouseArgs : IInputEventArgs
    {
        /// <summary>
        /// Mouse button as SpaceVIL.Core.MouseButton.
        /// </summary>
        public MouseButton Button;

        /// <summary>
        /// State of input as SpaceVIL.Core.InputState. Values: Release, Press, Repeat.
        /// </summary>
        public InputState State;

        /// <summary>
        /// Used modifiers while mouse input.
        /// </summary>
        public KeyMods Mods;

        /// <summary>
        /// Mouse cursor position.
        /// </summary>
        public Position Position = new Position();

        public ScrollValue ScrollValue = new ScrollValue();
        
        /// <summary>
        /// Clearing MouseArgs
        /// </summary>
        public void Clear()
        {
            Button = MouseButton.Unknown;
            State = 0;
            Mods = 0;
            Position.SetPosition(0, 0);
            ScrollValue.SetValues(0, 0);
        }

        public override String ToString()
        {
            return
                Button + " " +
                State + " " +
                Mods + " " +
                ScrollValue + " " +
                Position.GetX() + " " +
                Position.GetY() + " ";
        }
    }
}